### PR TITLE
Upgrade prometheus to 2.19.1

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
In this PR I'm upgrading Prometheus to `master` (and Thanos too) to primarily add two changes:
- Fix unsequential m-map files (https://github.com/prometheus/prometheus/pull/7414)
- Optimized bstream reader used by XORChunk iterator (https://github.com/prometheus/prometheus/pull/7390)

This is a **draft PR** because:
- Blocked by the Prometheus fix https://github.com/prometheus/prometheus/pull/7435

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/querier/lazyquery/lazyquery.go [hunk 1]
```diff
@@ -48,22 +48,17 @@ func (l LazyQuerier) createSeriesSet(selectSorted bool, params *storage.SelectHi
...
-			future <- set
-		}
+		future <- l.next.Select(selectSorted, params, matchers...)
 	}()
 	return future
 }
 
 // Select implements Storage.Querier
-func (l LazyQuerier) Select(selectSorted bool, params *storage.SelectHints, matchers ...*labels.Matcher) (storage.SeriesSet, storage.Warnings, error) {
+func (l LazyQuerier) Select(selectSorted bool, params *storage.SelectHints, matchers ...*labels.Matcher) storage.SeriesSet {
```
Comment (pstibrany) [inline pkg/querier/lazyquery/lazyquery.go:57]: We can get rid of `createSeriesSet` now. It was introduced when there were Select and SelectSorted methods.

File: pkg/querier/querier.go [hunk 2]
```diff
@@ -258,49 +258,38 @@ func (q querier) Select(_ bool, sp *storage.SelectHints, matchers ...*labels.Mat
...
-		return nil, nil, promql.ErrStorage{Err: err}
+		return storage.ErrSeriesSet(promql.ErrStorage{Err: err})
 	}
 
 	if len(q.queriers) == 1 {
-		seriesSet, warning, err := q.queriers[0].Select(true, sp, matchers...)
-		if err != nil {
-			return nil, warning, err
-		}
+		seriesSet := q.queriers[0].Select(true, sp, matchers...)
```
Comment (pstibrany) [inline pkg/querier/querier.go:270]: This changes the logic slightly, but end result looks to be the same.

Comment (pracucci) [reply pkg/querier/querier.go:270]: I checked the code and practically there should be no difference. The `seriesSet.Err()` had to be handled even before.

File: pkg/querier/lazyquery/lazyquery.go (hunk 1 above)
Comment (pracucci) [reply pkg/querier/lazyquery/lazyquery.go:57]: Good point, done.

**Pull request merged**